cmd/minikube/cmd/delete.go

```diff
@@ -496,6 +497,15 @@ func deleteProfileDirectory(profile string) {
 	}
```

Comment:
thanks for finding out this ! we were not cleaning up the machine folder for multi node


I wonder if there are other places that we are missing this

for example
in func demolish

https://github.com/medyagh/minikube/blob/75cd17801287339a00efe44fbd451b4dbfadef86/pkg/minikube/machine/delete.go#L122

is there a chance we could add this delete Machine directory to wherever it is doing for the single node ?

my undrestanding is, for single node delete the machine folder correctly but for multinode we are not.
is there any other place in the code we could do this ? (as opposed to a deffered clean up?)

Reply:
Even for the single node, it looks `DeleteHost` not  remove directory.
It's handled by `deleteProfileDirectory`
https://github.com/kubernetes/minikube/blob/f68005724c95b67d4002a0612291c655ec5b921f/cmd/minikube/cmd/delete.go#L330-L331

```
❯ minikube start -p test-profile
😄  [test-profile] minikube v1.16.0 on Darwin 10.15.7
✨  Automatically selected the docker driver
👍  Starting control plane node test-profile in cluster test-profile
🔥  Creating docker container (CPUs=2, Memory=1988MB) ...
🐳  Preparing Kubernetes v1.20.0 on Docker 20.10.0 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "test-profile" cluster and "default" namespace by default
❯ minikube delete -p test-profile
🔥  Deleting "test-profile" in docker ...
🔥  Deleting container "test-profile" ...
🔥  Removing /Users/daehyeok/.minikube/machines/test-profile ...
💀  Removed all traces of the "test-profile" cluster.
```

`🔥  Removing /Users/daehyeok/.minikube/machines/test-profile ...` this line printed by `deleteProfileDirectory`
https://github.com/kubernetes/minikube/blob/f68005724c95b67d4002a0612291c655ec5b921f/cmd/minikube/cmd/delete.go#L488-L491

Reply:
maybe we should fix `delete` function ?
https://github.com/medyagh/minikube/blob/75cd17801287339a00efe44fbd451b4dbfadef86/pkg/minikube/machine/delete.go#L104

Reply:
that is a good idea ! @daehyeok .
admitably our delete code is the messiest and worst part of minikube code (huge huge huge chance for BIG refactor)
once we had an idea to break down the delete code in the CMD package into package.